src/server.c

```diff
@@ -3241,7 +3241,9 @@ static void propagateNow(int dbid, robj **argv, int argc, int target) {
 
     /* This needs to be unreachable since the dataset should be fixed during
      * replica pause (otherwise data may be lost during a failover) */
-    serverAssert(!(isPausedActions(PAUSE_ACTION_REPLICA) && (!server.client_pause_in_transaction)));
+    int is_pause_replica_action = isPausedActions(PAUSE_ACTION_REPLICA);
+    int is_pause_shutdown_purpose = isPausedPurpose(PAUSE_DURING_SHUTDOWN);
+    serverAssert(!(is_pause_replica_action && !is_pause_shutdown_purpose && !server.client_pause_in_transaction));
```

Comment:
This was hard for me to read. This simplification is equivalent but easier to read, or wdyt?

```suggestion
    serverAssert(!is_pause_replica_action || is_pause_shutdown_purpose || server.client_pause_in_transaction);
```

Please update the comment about this assert to, to make it easier to understand the extra cases about shutdown and in-transaction.

Reply:
yes, the new code is better to understand, i updated the code and the comment.